### ISSUE TITLE
Download multiple files in parallel

### ISSUE THREAD

**Issue by rollsch** [microsoft/AzureStorageExplorer#7584]
### Preflight Checklist

- [X] I have installed the [latest version of Storage Explorer](https://github.com/Microsoft/AzureStorageExplorer/releases/latest).
- [X] I have checked existing resources, including the [troubleshooting guide](https://docs.microsoft.com/azure/storage/common/storage-explorer-troubleshooting) and the [release notes](https://github.com/microsoft/AzureStorageExplorer/releases).
- [X] I have searched for [similar issues](https://github.com/Microsoft/AzureStorageExplorer/issues).

### Problem

I would like to download files in parallel. Eg if Iam downloading a 1gb zip, then I want to download a 1 kb log file I have to wait. What is stopping both from downloading in parallel? 

### Desired Solution

I would like to download files in parallel via the UI. Eg click on one file, then click on a second file before the first has finished without it queuing up the file.

### Alternatives and Workarounds

I can open a second instance of azure storage explorer, however my quick access menu is missing, this makes it time consuming to locate my files.

### Additional Context

Can I publicly contribute to implementing said feature? If so where in the code base should I start reading/researching. I am happy to beta test, or even have it simply launch azcopy in a separate process with no UI feedback at all. It doesn't need to be pretty.

**Comment by craxal** [comment]
@rollsch File downloads are considered "large" tasks, and we limit large tasks to one at a time. We can consider increasing this limit, but others might have machines that have trouble sustaining lots of big tasks.

**Comment by JasonYeMSFT** [comment]
Let's investigate allowing concurrent execution of AzCopy jobs in 1.34.0.

**Comment by JasonYeMSFT** [comment]
I'll just bump the job limit to 5 for now. As a remainder, Storage Explorer currently offers 3 capabilities to avoid running into issues with concurrent AzCopy jobs.
- "Auto" concurrency setting
- Configurable per-job bandwidth limitation
- Cancelable pending AzCopy jobs

If you run into issues such as AzCopy using too much network bandwidth, concurrent jobs contending for CPU cycles, low priority pending jobs blocking high priority jobs, etc., consider using these features to work around them

**Comment by rollsch** [comment]
Can wevadd this feature and simply set the default to 1?

**Comment by JasonYeMSFT** [comment]
> Can we add this feature and simply set the default to 1?

We will bump the job limit to 5 in 1.34.0. In other words, you will be able to have up to 5 concurrent AzCopy copy jobs starting from 1.34.0.

**Comment by rollsch** [comment]
Fantastic. Any reason to not make it configurable?

**Comment by MRayermannMSFT** [comment]
> Fantastic. Any reason to not make it configurable?

We are worried about users not understanding how resource intensive AzCopy can be, doing too much at once, and ultimately having a terrible user experience. We'll start with 5 and go from there. :)

**Comment by rollsch** [comment]
Could you make it a hidden/hard to find option? My bigggest grief is if I download 5 files that are several gb, then i want to view a file that is 1kb i cannot. Opening a second copy of storage explorer loses my quickaccess menu (I can understand why this is a problem due to concurrent access ) however I feel it should be up to the user to have a terrible experience if they want. Maybe a simple red text warning label would be sufficient. Many of us have > Gbps up/down and i struggle to see how bad it could get.

**Comment by craxal** [comment]
For a not insignificant number of users, it can get bad (see #1099, for example). If turning up the limit to 5 turns out to be detrimental to users, we may have to dial it back down. If this is the case, we can consider making this configurable.

Until that time, do you find yourself downloading many large files and wanting to preview small files often? Are you able to work around that by previewing the small file before starting the big downloads? Or, if you need to do more intensive download work, is to use AzCopy directly, which will give you direct control, leaving Storage Explorer open for previewing.
